scrapers_next/ak/people.py

```diff
@@ -1,69 +1,143 @@
-from spatula import XmlListPage, URL, XPath
+import re
+import attr
+from spatula import HtmlPage, HtmlListPage, XPath, CSS, SelectorError
 from openstates.models import ScrapePerson
 
 
-ELEMENTS = (
-    "FirstName",
-    "MiddleName",
-    "LastName",
-    "EMail",
-    "Phone",
-    "District",
-    "Party",
-    "Building",
-    "Room",
-)
+@attr.s(auto_attribs=True)
+class PartialMember:
+    url: str
+    chamber: str = ""
 
 
-def _get_if_exists(item, elem):
-    val = item.xpath(f"./{elem}/text()")
-    if val:
-        return str(val[0])
+class LegDetail(HtmlPage):
+    example_source = "http://www.akleg.gov/basis/Member/Detail/32?code=BCH"
 
+    def process_page(self):
 
-class Legislators(XmlListPage):
-    session_num = "32"
-    source = URL(
-        "http://www.legis.state.ak.us/publicservice/basis/members"
-        f"?minifyresult=false&session={session_num}",
-        headers={"X-Alaska-Legislature-Basis-Version": "1.4"},
-    )
-    selector = XPath("//Member/MemberDetails")
+        details_div = CSS(".bioright").match(self.root)[0]
 
-    def process_item(self, item):
-        item_dict = {elem: _get_if_exists(item, elem) for elem in ELEMENTS}
-        chamber = item.attrib["chamber"]
-        code = item.attrib["code"].lower()
-
-        party = item_dict["Party"]
-        if party == "N":
-            party = "Independent"
-
-        person = ScrapePerson(
-            name="{FirstName} {LastName}".format(**item_dict),
-            given_name=item_dict["FirstName"],
-            family_name=item_dict["LastName"],
+        name_span = CSS(".formal_name").match(details_div)[0].text_content()
+        name_list = name_span.split(" ")
+        given_name = name_list[1]
+        family_name = " ".join(name_list[2:])
+
+        email = CSS("a").match(details_div)[0].text_content().strip()
+
+        div_text = details_div.text_content().replace("\r\n", " ")
+
+        details = {}
+
+        detail_patterns = [
+            "District",
+            "Party",
+            "Toll-Free",
+            "Phone",
+            "Fax",
+        ]
+
+        for pattern in detail_patterns:
+            match = re.search(rf"{pattern}:\s\S+", div_text)
+            if match:
+                detail = div_text[match.start() + len(pattern) + 2 : match.end()]
+                details[pattern] = detail
+
+        address_patterns = [r"Interim.+99\d\d\d", r"Session.+99801"]
```

Comment:
May want to rethink this pattern, just in case of cases where both the session & interim addresses are ending with the 99801 zip code like [Senator Kiehl](https://www.akleg.gov/basis/Member/Detail/32?code=KIE)

Reply:
The [newest commit has address pattern handling broken up](https://github.com/openstates/openstates-scrapers/pull/4156/commits/c6ab03807dad7b26ae8f8f3a2f5941177dea3fff#diff-fb9ba31d11ce10e77c56b79a04bbc75d0b174873ef40566f473109efb522f0f0R49-R63), which more effectively deals with cases like Senator Kiehl.

I also made the regex checks more elegant, and built a separate `format_address()` [helper function](https://github.com/openstates/openstates-scrapers/pull/4156/commits/c6ab03807dad7b26ae8f8f3a2f5941177dea3fff#diff-fb9ba31d11ce10e77c56b79a04bbc75d0b174873ef40566f473109efb522f0f0R7-R13) to ensure addresses have:
- collapsed whitespace
- a comma between city and state
- a comma between address lines (i.e. "PO Box 844 201 Katlian Street" becomes "PO Box 844, 201 Katlian Street")